### PR TITLE
Add drag'n'drop feature

### PR DESCRIPTION
**Pull request by FelixBaensch**

### CODE REVIEW

**Comment by FelixBaensch** [comment]
See #33

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java [hunk 1]
```diff
@@ -384,6 +386,22 @@ else if (keyEvent.getCode() == KeyCode.LEFT || keyEvent.getCode() == KeyCode.PAG
             this.mainView.getMainMenuBar().getHistogramViewerMenuItem().setDisable(newValue.getId().equals(TabNames.MOLECULES.toString()));
             this.mainView.getMainMenuBar().getOverviewViewMenuItem().setDisable(newValue.getId().equals(TabNames.ITEMIZATION.toString()));
         }));
+        this.mainView.getMainCenterPane().setOnDragOver(aDragEvent -> {
+            if (aDragEvent.getGestureSource() != mainView.getMainCenterPane() && aDragEvent.getDragboard().hasFiles()){
```
Comment (JonasSchaub) [inline src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java:390]: "this" statement missing for mainView.

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java [hunk 2]
```diff
@@ -384,6 +386,22 @@ else if (keyEvent.getCode() == KeyCode.LEFT || keyEvent.getCode() == KeyCode.PAG
...
+                aDragEvent.acceptTransferModes(TransferMode.COPY);
+            }
+            aDragEvent.consume();
+        });
+        this.mainView.getMainCenterPane().setOnDragDropped(aDragEvent -> {
+            Dragboard tmpDragboard = aDragEvent.getDragboard();
+            boolean tmpSucceeded = false;
+            if (tmpDragboard.hasFiles()) {
+                importMoleculeFile(tmpDragboard.getFiles().getFirst());
```
Comment (JonasSchaub) [inline src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java:399]: "this" statement missing.

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java [hunk 3]
```diff
@@ -440,10 +458,11 @@ private void closeWindowEvent(WindowEvent anEvent) {
         anEvent.consume();
     }
     //
+
     /**
-     * Loads molecule file and opens molecules tab.
+     * Opens a file choose, loads the chosen file and opens molecules tab
      *
-     * @param aParentStage Stage where to open the file chooser dialog
+     * @param aParentStage aParentStage Stage where to open the file chooser dialog
```
Comment (JonasSchaub) [inline src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java:465]: Why doubling the param name?

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java [hunk 4]
```diff
@@ -440,10 +458,11 @@ private void closeWindowEvent(WindowEvent anEvent) {
...
      */
     private void importMoleculeFile(Stage aParentStage) {
```
Comment (JonasSchaub) [inline src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java:467]: Maybe rename this method to make the difference more clear. "chooseAndImportMoleculeFile()"?

**Review by JonasSchaub — CHANGES_REQUESTED** [review]

**Comment by JonasSchaub** [comment]
Looks nice and also works under Windows! 👍 

I don't want to overburden you, but would it be possible to add a file extension filter? Right now, if I drag and drop e.g. a PDF file into MORTAR, it starts importing and then tells me "import failed". Also an attempted multi-file drag and drop could be prohibited.

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java (hunk 4 above)
Comment (FelixBaensch) [reply src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java:467]: Therefore the javadoc, but if it makes you happy

File: src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java (hunk 3 above)
Comment (FelixBaensch) [reply src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java]: Copy & Paste before the second coffee

**Review by JonasSchaub — APPROVED** [review]

**Pull request merged**